experimental/devices/bmp180/bmp180.go

```diff
@@ -0,0 +1,265 @@
+// Copyright 2017 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+// Package bmp180 controls a Bosch BMP180 device over I²C.
+//
+// Datasheet
+//
+// The official data sheet can be found here:
+//
+// https://ae-bst.resource.bosch.com/media/_tech/media/datasheets/BST-BMP180-DS000-121.pdf
+//
+// The font the official datasheet on page 15 is unreadable, a copy with
+// readable text can be found here:
+//
+// https://cdn-shop.adafruit.com/datasheets/BST-BMP180-DS000-09.pdf
+//
+// Notes on the datasheet
+//
+// The results of the calculations in the algorithm on page 15 are partly
+// wrong. It looks like the original authors used non-integer calculations and
+// some nubers were rounded. Take the results of the calculations with a grain
+// of salt.
+package bmp180
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"time"
+
+	"periph.io/x/periph/conn/i2c"
+	"periph.io/x/periph/conn/mmr"
+	"periph.io/x/periph/devices"
+)
+
+// Oversampling affects how much time is taken to measure pressure.
+type Oversampling uint8
+
+// Possible oversampling values.
+const (
+	No  Oversampling = 0
+	O2x Oversampling = 1
+	O4x Oversampling = 2
+	O8x Oversampling = 3
+)
+
+// bit offsets in regCtrlMeas
+const (
+	ctrlMeasurementControlShift = 0
+	ctrlStartConversionShift    = 5
+	ctrlOversamplingShift       = 6
+)
```

Comment:
FTR, the reason is that I care about vertical density; the more dense the code can be vertically, the more a screenful page shows more code, leading to better visualization. Obviously this is a trade off with readability, as some people need more vertical spacing than others but I feel it is generally abused so I prefer not too many empty lines.